src/memray/_memray/hooks.cpp

```diff
@@ -205,7 +205,11 @@ calloc(size_t num, size_t size) noexcept
 {
     assert(hooks::calloc);
 
-    void* ret = hooks::calloc(num, size);
+    void* ret;
```

Comment:
Yeah, the commit message for that one is:
```
Ignore allocations under calloc and posix_memalign

Apparently Alpine implements `calloc` in terms of `malloc`, and
`posix_memalign` in terms of `aligned_alloc`. Ensure that we don't
record duplicate allocations on Alpine.

Our `ALL_ALLOCATIONS` capture file format incorrectly recorded both the
`calloc` and the `malloc` it is implemented in terms of, but because the
`malloc` happens first and returns the same pointer as the `calloc`
later returns, the aggregator wound up overwriting the allocation record
for the `malloc` with the one for the `calloc`, silently dropping it.
```

I should maybe have made this a separate PR. It's an existing bug that our existing tests didn't catch, but tests introduced on this PR did.